crypto/dh/dh_pmeth.c

```diff
@@ -372,22 +366,17 @@ static int pkey_dh_paramgen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey)
             return 0;
         evp_pkey_set_cb_translate(pcb, ctx);
     }
-#ifndef OPENSSL_NO_DSA
+# ifdef FIPS_MODE
+    dctx->use_dsa = 2;
```

Comment:
The use_dsa value of "2" does not seem to be documented:

> The EVP_PKEY_CTX_set_dh_paramgen_type() macro sets the key type for DH
> parameter generation. Use 0 for PKCS#3 DH and 1 for X9.42 DH.
> The default is 0.

Maybe it might be nice to use macros rather than "magic" numbers?

Side note: `use_dsa` seems like a terrible name

Reply:
The name is mapping to what it does in the app...
i.e- dsa has extra fields in ANS1 that dh doesnt have - so it makes sense to me.

Reply:
Changed name to paramgen_type and added defines..